ml-agents/mlagents/trainers/settings.py

```diff
@@ -151,6 +155,102 @@ class CuriositySettings(RewardSignalSettings):
     learning_rate: float = 3e-4
 
 
+class ParameterRandomizationType(Enum):
+    UNIFORM: str = "uniform"
+    GAUSSIAN: str = "gaussian"
+    MULTIRANGEUNIFORM: str = "multirangeuniform"
+
+    def to_settings(self) -> type:
+        _mapping = {
+            ParameterRandomizationType.UNIFORM: UniformSettings,
+            ParameterRandomizationType.GAUSSIAN: GaussianSettings,
+            ParameterRandomizationType.MULTIRANGEUNIFORM: MultiRangeUniformSettings,
+        }
+        return _mapping[self]
+
+
+@attr.s(auto_attribs=True)
+class ParameterRandomizationSettings:
+    seed: int = parser.get_default("seed")
+
+    @staticmethod
+    def structure(d: Mapping, t: type) -> Any:
+        """
+        Helper method to structure a Dict of ParameterRandomizationSettings class. Meant to be registered with
+        cattr.register_structure_hook() and called with cattr.structure(). This is needed to handle
+        the special Enum selection of ParameterRandomizationSettings classes.
+        """
+        if not isinstance(d, Mapping):
+            raise TrainerConfigError(
+                f"Unsupported parameter randomization configuration {d}."
+            )
+        d_final: Dict[str, List[float]] = {}
+        for param, param_config in d.items():
+            if param == "resampling-interval":
+                logger.warning(
+                    "The resampling-interval is no longer necessary for parameter randomization. It is being ignored."
+                )
+                continue
+            if not isinstance(param_config, Mapping):
+                raise TrainerConfigError(
+                    f"Unsupported distribution configuration {param_config}."
+                )
+            for key, val in param_config.items():
+                enum_key = ParameterRandomizationType(key)
+                t = enum_key.to_settings()
+                d_final[param] = strict_to_cls(val, t)
+        return d_final
+
+
+@attr.s(auto_attribs=True)
+class UniformSettings(ParameterRandomizationSettings):
+    min_value: float = attr.ib()
+    max_value: float = 1.0
+
+    @min_value.default
+    def _min_value_default(self):
+        return 1.0
+
+    @min_value.validator
+    def _check_min_value(self, attribute, value):
+        if self.min_value > self.max_value:
+            raise TrainerConfigError(
+                "Minimum value is greater than maximum value in uniform sampler."
+            )
+
+
+@attr.s(auto_attribs=True)
+class GaussianSettings(ParameterRandomizationSettings):
+    mean: float = 1.0
+    st_dev: float = 1.0
+
+
+@attr.s(auto_attribs=True)
+class MultiRangeUniformSettings(ParameterRandomizationSettings):
+    intervals: List[List[float]] = attr.ib()
```

Comment:
This seems inconsistent with https://github.com/Unity-Technologies/ml-agents/pull/4065/files#diff-030db66da9504a24544d1979beb4248eR85

Reply:
I updated the function as per your comment above.